Tests/MongoSwiftTests/UnifiedTestRunner/UnifiedTestRunner.swift

```diff
@@ -175,46 +197,47 @@ struct UnifiedTestRunner {
                     entities: try file.createEntities?.toEntityMap() ?? [:],
                     internalClient: self.internalClient
                 )
-
                 // Workaround for SERVER-39704:  a test runners MUST execute a non-transactional distinct command on
                 // each mongos server before running any test that might execute distinct within a transaction. To ease
                 // the implementation, test runners MAY execute distinct before every test.
                 if self.topologyType.isSharded && !MongoSwiftTestCase.serverless {
                     let collEntities = context.entities.values.compactMap { try? $0.asCollection() }
-                    for (_, client) in try self.internalClient.asMongosClients() {
+                    let clientMap = try self.internalClient.asMongosClients()
+                    for (_, client) in clientMap {
                         for entity in collEntities {
-                            _ = try client.db(entity.namespace.db).runCommand(
-                                ["distinct": .string(entity.name), "key": "_id"]
-                            )
+                            do {
+                                _ = try await client.db(entity.namespace.db).runCommand(
+                                    ["distinct": .string(entity.name), "key": "_id"]
+                                )
+                            } catch {
+                                // DB not explicitly initially created for operation-failure test bc no initialData
+                                if !(error is MongoError.CommandError) {
```

Comment:
we can make this a little more specific actually, so that we don't ignore arbitrary command errors. the [error code](https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.yml#L63) for `NamespaceNotFound` is 26. 

let's do something like the logic here so that we only catch and ignore command errors with code 26:
https://github.com/mongodb/mongo-swift-driver/blob/2d579e2002e2c80aab8fcdc2159d1ffad79f6147/Tests/MongoSwiftSyncTests/UnifiedTestRunner/UnifiedTestRunner.swift#L76